train_spacy.py

```diff
@@ -0,0 +1,55 @@
+import spacy
+from spacy.training.example import Example
+from spacy.language import Language
+
+
+def load_training_data(file_path: str) -> list:
+    """Load training data from a text file.
+    Data is formatted as text, followed by a newline,
+    followed by a list of names separated by newlines.
+    Each entry is separated by two newlines.
+    """
+
+    with open(file_path, encoding="utf-8") as file:
+        data = file.read().split("\n\n")
+        data = [item.split("\n") for item in data]
+        return data
+
+
+def _format_training_data(data: list) -> list:
+    """Format training data for spacy."""
+    formatted_data = []
+    for item in data:
+        text = item[0]
+        entities = []
+        for name in item[1:]:
+            start = text.find(name)
+            end = start + len(name)
+            entities.append((start, end, "PERSON"))
+        formatted_data.append((text, {"entities": entities}))
+    return formatted_data
+
+
+def train_model(data: list) -> Language:
```

Comment:
Since in our case, the caller has already loaded model(s), let the caller pass the model into `train_model()` instead of hard-coding it.